draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -537,447 +539,170 @@ Note: Examples in this document use `cnf` Claim defined in [@RFC7800] to include
-
-## Processing Model {#processing_model}
-
-Neither an SD-JWT nor an HS-Disclosures JWT is suitable for direct use by an application.
-Besides the REQUIRED verification steps listed above, it is further RECOMMENDED
-that an application-consumable format is generated from the data released in
-the HS-Disclosures. The RECOMMENDED way is to merge the released claims and any
-plaintext claims in the SD-JWT recursively:
-
- * Objects from the released claims must be merged into existing objects from the SD-JWT.
- * If a key is present in both objects:
-   * If the value in the released claims is an object and the value in the
-     SD-JWT claims is an object, the two objects MUST be merged recursively.
-   * Else, the value in the released claims MUST be used.
-
-The keys `sd_digests` and `sd_digest_derivation_alg` SHOULD be removed prior to further
-processing.
-
-The processing is shown in Examples 2b and 3 in the Appendix.
+    5. Check that the `sd_digest_derivation_alg` claim is present and its value is understood and the digest derivation algorithm is deemed secure.
+ 4. Create a copy of the SD-JWT payload for further processing, if required.
+ 5. Process the Disclosures. For each Disclosure provided:
+    1. Calculate the hash digest over the base64url string as described in (#hashing_disclosures).
+    2. Find any `_sd` keys in the SD-JWT payload where the digest of the Disclosure is included in the array.
+       1. If the digest cannot be found in the SD-JWT payload, the Verifier MUST reject the Presentation.
+       2. If there is more than one place where the digest is included, the Verifier MUST reject the Presentation.
+       3. If there is a key `_sd` that does not refer to an array, the Verifier MUST reject the Presentation.
+       4. Otherwise, insert, at the level of the `_sd` claim, the claim described by the Disclosure with the claim name and claim value provided in the Disclosure.
+          1. If the Disclosure is not a JSON-encoded array of three elements, the Verifier MUST reject the Presentation.
+          2. If the claim name already exists, the Verifier MUST reject the Presentation. Note that this also means that if a Holder sends the same Disclosure multiple times, the Verifier MUST reject the Presentation.
+          3. If the claim value is an object and contains an `_sd` key, the Verifier MUST reject the Presentation.
+    3. Remove all `_sd` claims from the SD-JWT payload.
+    4. Remove the claim `sd_digest_derivation_alg` from the SD-JWT payload.
+ 6. If Holder Binding is required:
+    1. If Holder Binding is provided by means not defined in this specification, verify the Holder Binding according to the method used.
+    2. Otherwise, verify the Holder Binding JWT as follows:
+       1. If not Holder Binding JWT is provided, the Verifier MUST reject the Presentation.
+       2. Determine the public key for the Holder from the Holder Binding JWT.
```

Comment:
A step that the public key signing the Holder Binding JWT must be included in the SD-JWT signed by the Issuer needs ot be added.